File: src/generic/FileSelector.java
package generic;

import java.io.File;
import java.util.Scanner;

public class FileSelector {

    public static String getFilePath(String filePath) {
        if (!filePath.isEmpty())
            return filePath;

        File file;
        do {
            System.out.print("Enter a valid data set file path: ");
            Scanner scanner = new Scanner(System.in);
            file = new File(scanner.nextLine());
        } while(!file.exists());

        return file.getPath();
    }
}

File: src/python/fileselector.py
import os


class FileSelector:
    @staticmethod
    def get_file_path(file_path):
        if file_path:
            return file_path

        while True:
            input_path = input("Enter a valid data set file path: ")
            if os.path.exists(input_path):
                return input_path
